internal/pkg/deploy/cloudformation/cloudformation.go

```diff
@@ -576,7 +576,9 @@ func (cf CloudFormation) errOnFailedStack(stackName string) error {
 	}
 	status := aws.StringValue(stack.StackStatus)
 	if cloudformation.StackStatus(status).IsFailure() {
-		return fmt.Errorf("stack %s did not complete successfully and exited with status %s", stackName, status)
+		return fmt.Errorf("stack %s did not complete successfully and exited with status %s.\n"+
```

Comment:
instead of a multi-line error - could we return an error struct with `RecommendActions()` for the follow up steps? an example: https://github.com/aws/copilot-cli/blob/ea86849093b9ffb03a8efec4cb595ad4fd10fc4a/internal/pkg/cli/svc_deploy.go#L555-L589

Reply:
I think we're better off keeping it as a multi-line error for now. When we move to stage 3 of Failed Service Management where we have error codes (and error messages), we would then likely need to make the recommended actions struct. For now, it seems like a straightforward static error message would be sufficient and clearer.

Let me know if you'd still like me to make the RecommendedActions struct for this change

Reply:
Included struct.

Reply:
awesome! just wanted to clarify the implementation - your struct should have two functions on it: `Error() string` and `RecommendActions() string` 😊. The `Error()` function returns a short, one line error, and the `RecommendActions()` function returns to more detailed instructions.

Reply:
+1 to Danny comment. Something like this ⬇️ 
```
type errFailedService struct {
    stackName string
    status    string
}
func (e *errFailedService) RecommendActions() string {
    return log.Infof("You may fix the error by updating the service code or the manifest configuration.\n"+
        "You can then retry deploying your service by running %s.",color.HighlightCode("copilot svc deploy"))
}
func (e *errFailedService) Error() error {
    return fmt.Errorf("stack %s did not complete successfully and exited with status %s.", e.stackName, e.status)
}
if cloudformation.StackStatus(status).IsFailure() {
    return &errFailedService{
        stackName: stackName,
        status: status
    }
}
```